### PR TITLE
Add an option to docker and build script to allow mvn to run all tests

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

File: dev/jenkins/build.sh [hunk 1]
```diff
@@ -20,6 +20,12 @@ then
   git clean -fdx
 fi
 
+local mvn_args=""
+if [ -n "${ALLUXIO_MVN_RUNTOEND}" ]
+then
+  mvn_args+=" -fn --fail-at-end"
```
Comment (Xenorith) [inline dev/jenkins/build.sh:26]: no need leading space in string

Comment (yuzhu) [reply dev/jenkins/build.sh:26]: it is not necessary, but this makes it easier to copy/paste later?

Comment (Xenorith) [reply dev/jenkins/build.sh:26]: true that, and there's no harm in having the extra space

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Pull request merged**